cmd/deploy/deploy.go

```diff
@@ -618,3 +595,82 @@ func (dc *DeployCommand) cleanUp(ctx context.Context) {
 		oktetoLog.Infof("could not stop local server: %s", err)
 	}
 }
+
+func (dc *DeployCommand) buildImages(ctx context.Context, cfg *corev1.ConfigMap, c kubernetes.Interface, pipelineData *pipeline.CfgData, deployOptions *Options) error {
+	if deployOptions.Build {
+		buildOptions := &types.BuildOptions{
+			EnableStages: true,
+			Manifest:     deployOptions.Manifest,
+			CommandArgs:  deployOptions.servicesToDeploy,
+		}
+		oktetoLog.Debug("force build from manifest definition")
+		if errBuild := dc.Builder.Build(ctx, buildOptions); errBuild != nil {
+			return updateConfigMapStatusError(ctx, cfg, c, pipelineData, errBuild)
+		}
+	} else {
+		stack := deployOptions.Manifest.GetStack()
+		if stack == nil {
```

Comment:
This causes an odd behavior if a manifest has a build section and doesn't have a stack to exist without. Adding a scenario to help you debug the issue (remember to create a new namespace to run the command):

1. Create the following resources
```yaml
build:
  app:
    context: .
deploy:
  - ls
```
```Dockerfile
FROM alpine
```
2. Run the following command: `okteto deploy --namespace new-namespace`
3. See that no `okteto stack` has been created

Reply:
Can we add a unit test to the new function mocking the registry and the builder so we can avoid this issue to happen again? If you think the PR has a lot of changes, we can leave it as technical debt (creating the proper issue) and address it when this PR is merged

Reply:
@jLopezbarb I've added unit tests to verify the behavior